openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/OpenApiInfoUpdaterTask.java

```diff
@@ -172,11 +172,20 @@ private SeparatedNodeList<MappingFieldNode> getUpdatedFields(AnnotationNode anno
         MappingConstructorExpressionNode annotationValue = annotationValueOpt.get();
         SeparatedNodeList<MappingFieldNode> existingFields = annotationValue.fields();
         Token separator = NodeFactory.createToken(SyntaxKind.COMMA_TOKEN);
+        boolean openApiDefAvailable = false;
         for (MappingFieldNode field : existingFields) {
+            if (field instanceof SpecificFieldNode) {
+                String fieldName = ((SpecificFieldNode) field).fieldName().toString();
+                openApiDefAvailable = Constants.OPEN_API_DEFINITION_FIELD.equals(fieldName);
+            }
             fields.add(field);
             fields.add(separator);
         }
-        fields.add(createOpenApiDefinitionField(servicePath));
+        if (openApiDefAvailable) {
+            fields.remove(fields.size() - 1);
```

Comment:
No, I don't think field size will be zero. Because we are checking whether annotation have fields or not in line 167-170.
```java
Optional<MappingConstructorExpressionNode> annotationValueOpt = annotation.annotValue();
if (annotationValueOpt.isEmpty()) {
    return NodeFactory.createSeparatedNodeList(createOpenApiDefinitionField(servicePath));
}
```

Reply:
Yeah no issue then 🙂 , sorry I have missed that lines.

Reply:
Improved the logic with [8dc4bf6485d37fbdbdae681d053b79bd3bced0df](https://github.com/ballerina-platform/openapi-tools/pull/966/commits/8dc4bf6485d37fbdbdae681d053b79bd3bced0df)